### PR TITLE
8289384: Fix warnings: method does not override the inherited method since it is private to a different package

### PR DESCRIPTION
**Pull request by andy-goryachev-oracle**
The fix includes:
- renaming of offending methods to avoid confusion
- explicitly declaring the offending methods as private

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8289384](https://bugs.openjdk.org/browse/JDK-8289384): Fix warnings: method does not override the inherited method since it is private to a different package

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/824/head:pull/824` \
`$ git checkout pull/824`

Update a local copy of the PR: \
`$ git checkout pull/824` \
`$ git pull https://git.openjdk.org/jfx pull/824/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 824`

View PR using the GUI difftool: \
`$ git pr show -t 824`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/824.diff">https://git.openjdk.org/jfx/pull/824.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.graphics/src/main/java/com/sun/prism/impl/ps/BaseShaderGraphics.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ protected BaseShaderGraphics(BaseShaderContext context,
         this.context = context;
     }
 
-    BaseShaderContext getContext() {
+    BaseShaderContext getBaseShaderContext() {
         return context;
     }
```
Comment (nlisker) [inline modules/javafx.graphics/src/main/java/com/sun/prism/impl/ps/BaseShaderGraphics.java:92]: Mentioning this was brought up in relation to a different PR: https://github.com/openjdk/jfx/pull/628#issuecomment-924898354

Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/src/main/java/com/sun/prism/impl/ps/BaseShaderGraphics.java:92]: I think it's a safe change (and this method should probably be made private, as it is only called from the same class).

Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/com/sun/prism/impl/ps/BaseShaderGraphics.java:92]: > I think it's a safe change

Agreed.

> (and this method should probably be made private, as it is only called from the same class).

No, it's called from `CachingShapeRep`, which you (necessarily) modified as part of this PR.

Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/src/main/java/com/sun/prism/impl/ps/BaseShaderGraphics.java:92]: right, sorry - that's why I did not make it private.

(note to self: double check before responding)

**Comment by kevinrushforth** [comment]
One question: did you check that all of the methods you changed to private scope are only called from other methods in the same class where they are defined (and not called from another class, such as an inner class or lambda, in the same file)?

**Comment by andy-goryachev-oracle** [comment]
> One question: did you check that all of the methods you changed to private scope are only called from other methods in the same class where they are defined (and not called from another class, such as an inner class or lambda, in the same file)?

yes, of course.   besides, all these cases would generate an error or a warning in eclipse.

**Comment by andy-goryachev-oracle** [comment]
Kevin, what I meant actually, is that Eclipse would generate a warning when compiler creates a synthetic accessor:

Access to enclosing method beginChange() from the type ObservableListBase<E> is emulated by a synthetic accessor method	line 59	/base/src/main/java/com/sun/javafx/collections	ElementObservableListDecorator.java

there are about 81 warnings like this, if enabled.  I decided to disable them because it may not be a trivial change.

Your snippet, of course, compiles without a problem, as it should.

**Comment by kevinrushforth** [comment]
Thanks for the explanation. I hadn't realized the Eclipse compiler had such a warning.

**Review by kevinrushforth — APPROVED** [review]
Looks good to me.

@nlisker any concerns?

**Comment by nlisker** [comment]
I haven't had time to look at this one yet. Will do so later today or tomorrow.

**Comment by nlisker** [comment]
I've finished an analysis of a few of the cases here. After checking the [JLS](https://docs.oracle.com/javase/specs/jls/se18/html/jls-9.html#jls-9.6.4.4), in my opinion this warning should be turned off without "fixing" its occurrences for the following reason:
If a method in a supertype is not visible to a subtype, then it is an implementation detail. Subtypes should not be concerned with the names of methods outside of their visibility (whether in the hierarchy or not). If a method is supposed to override another method, then the `@Override` annotation should be used. This will give the definitiveness of overriding - if the method really is overriding, nothing else is needed; if it isn't, a compiler error occurs.

Therefore, my recommendation is to fix all "method x is overriding method y but does not have the `@Override` annotation" warnings (I have my Eclipse set to treat missing override annotations as an error even, and it has served me well). Once all overriding methods are annotated and all non-overriding methods are not annotated, this warning becomes meaningless.

One interesting case is the one I mentioned with `D3DGraphics#getContext`: it really does override the superinterfce method, but also has the same signature of a method in `BaseShaderGraphics`, resulting in a "fake override", but the `@Override` annotation is still applicable. If we turn off the "fake override" warning, everything remains clear because the IDE tells us which method is really being overridden. In fact, this warning just made this case more confusing because it implies an override that doesn't happen, and hides the one that really does.

**Comment by kevinrushforth** [comment]
At a minimum it seems that this needs more discussion, which will need to wait until after the fork.

**Comment by andy-goryachev-oracle** [comment]
+1 for JLS

While I agree that leaving the code as is does not cause issues (since the methods are not getting overriden), it might fall under maintainability rubric.  Especially in the case of BaseShaderGraphics, where a human surely gets confused.  Some people do have eclipse and can F3 to see the superclass.

So I would suggest to merge all of my changes.

If this is unacceptable, at least merge the  BaseShaderGraphics change, as it is rather confusing.  

The good thing is that we only have a very few places, so the probability of confusion due to this issue is rather low.

**Comment by nlisker** [comment]
> These compiler warning settings are more than likely just a left over from the days before `@Override` existed.

My thoughts as well. `@Override` was added in 1.5. I think this warning was added prior.

**Comment by nlisker** [comment]
I turned on errors on missing `@Override` annotations. I got over 1k errors. I think it's worth fixing it in batches.

**Review by kevinrushforth — CHANGES_REQUESTED** [review]
Removing approval pending further discussion.

**Comment by andy-goryachev-oracle** [comment]
> I turned on errors on missing `@Override` annotations. I got over 1k errors. I think it's worth fixing it in batches.

missing @override is a different warning.  people do tend not to use this annotation, so any legacy code base usually contains lots of those.  same with synthetic accessors and unused imports.

I am ok with fixing all warnings, but people who use other IDEs or the IDEs with these warnings suppressed are bound to introduce them again and again.

**Comment by andy-goryachev-oracle** [comment]
@hjohn : 
I wholeheartedly agree.  Created JDK-8290244 to fix @override's at a later date, on a per-module basis.

**Comment by andy-goryachev-oracle** [comment]
I agree - we might want to leave synthetic methods as is and suppress the warning in Eclipse preferences, as the changes may not be trivial.

**Comment by andy-goryachev-oracle** [comment]
Kevin, what is the status of this PR?  It says "Changes requested" but I cannot find any specific requests.

**Comment by kevinrushforth** [comment]
On GitHub there is no way to dismiss an approval other than to submit a review with "Changes Requested", which I did. So there are no specific changes requested.

What is needed is to finish the review and answer the general questions that have been raised as to the direction we should go. There doesn't seem to be agreement that the code changes proposed in this PR are the ones we should be making, so let's continue the discussion. I don't think there is any urgency on this one, but let's see where the discussion goes.

**Comment by andy-goryachev-oracle** [comment]
Even though I would rather fix all the cases, especially BaseShaderGraphics, I am ok with turning the warning off and withdrawing this PR.

**Comment by andy-goryachev-oracle** [comment]
Looks like the general sentiment is to keep the code as is and disable the warning.  Withdrawing this PR.

**Pull request closed without merging**